WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -368,10 +368,12 @@ public static boolean isPostUploading(PostModel post) {
 
     public static void cancelQueuedPostUploadAndRelatedMedia(Context context, PostModel post) {
         if (post != null) {
-            PostUploadNotifier.cancelFinalNotification(context, post);
             if (sInstance != null) {
+                PostUploadNotifier.cancelFinalNotification(sInstance, post);
                 sInstance.mPostUploadNotifier.removePostInfoFromForegroundNotification(
                         post, sInstance.mMediaStore.getMediaForPost(post));
+            } else {
+                PostUploadNotifier.cancelFinalNotification(context, post);
```

Comment:
This will do nothing if `context` is null, so the notification won't be canceled. If that's a problem, perhaps the caller (`PostsListFragment`) should pass `WordPress.getContext()` rather than `getActivity()` as the context?

Reply:
I thought about it but wasn't sure - now that you mention it it gives me more confidence that we can use the app's  context here. Addressed in 6740780